earth_enterprise/src/fusion/autoingest/sysman/AssetOperation.cpp

```diff
@@ -129,7 +129,7 @@ void HandleTaskDone(const TaskDoneMsg & msg, MiscConfig::GraphOpsType graphOps)
         stateUpdater->SetSucceeded(version);
       }
       else {
-        stateUpdater->SetFailed(version);
+        stateUpdater->SetStateForRefAndDependents(msg.verref, AssetDefs::Failed, AssetDefs::Finished);
```

Comment:
Either this should still use SetFailed or we should remove SetFailed from the StateUpdater.

Reply:
Also, I think this will only set the state to Failed if the state is Finished (which means Failed, Succeeded, Canceled, Offline, or Bad). At this point the asset should be InProgress, so nothing will ever get set to failed.

Reply:
> Also, I think this will only set the state to Failed if the state is Finished (which means Failed, Succeeded, Canceled, Offline, or Bad). At this point the asset should be InProgress, so nothing will ever get set to failed.

I've now verified this through testing. I inserted code to randomly fail tasks in the resource provider, and Failed tasks are not marked appropriately. I think instead of `AssetDefs::Finished` we need to unconditionally return true. The third argument's purpose is to determine which assets in the dependency tree will be marked as Failed. Since none of them will return true for AssetDefs::Finished, none are marked as failed.

Reply:
Yeah, that was my fault for not looking closely at what the predicate was being used for. I'll update that.